### PR TITLE
Add tools for automated code cleanup

### LINKED ISSUE #963: Switch to Resharper Cleanup Code for code style
Use [Resharper Cleanup Code](https://www.jetbrains.com/help/resharper/CleanupCode.html) to automatically reformat code.

Motivation: 
- Avoid distraction from squiggles while writing/debugging code (StyleCop)
- Improve editor responsiveness
- Decrease local build time
- Automate whenever possible (to avoid discussions in PR feedback)
- Minimize the need for manual intervention to comply with desired code style

Tooling support to reformat files, based on the configured style:
- A key-press (Ctrl+E, F) in your IDE (Rider or Visual Studio with Resharper) on selected lines, file, project or solution
- A command-line tool to run locally, which takes an optional branch name. When passed, only changed files are reformatted
- An Azure DevOps Pipeline (activated using `/azp` command in PR) to make the cibuild reformat and commit
- An Azure DevOps Pipeline that validates whether all (changed) files match the configured style

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description

This PR adds tooling for automated code cleanup.
- An Azure DevOps pipeline to cleanup code, which is started using `/azp run cleanup-code` in a PR comment
- An Azure DevOps pipeline to validate code style, which runs as part of PR validation steps
- A PowerShell script to run locally, which takes an optional base branch name to reformat only changed files:
  ```
  C:\Source\Repos\Steeltoe [some-feature-branch ≡]> ./cleanupcode.ps1 main
  ```

Code cleanup performs the following steps, which are configured in src/Steeltoe.All.sln.DotSettings:

image

Fixes #963.

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [ ] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

**Comment by bart-vmware** [comment]
Note to reviewers: the VerifyCodeStyle task failing is to be expected: we haven't yet reformatted the codebase, so it correctly reports that the code does not comply with expected style.

File: build/pr-code-cleanup.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        exit 0
+  - task: PowerShell@2
+    displayName: Push changes
+    condition: and(succeeded(), eq(variables['hasChangesToCommit'], 'True'))
+    env:
+      SYSTEM_ACCESSTOKEN: $(System.AccessToken)  
+    inputs:
+      targetType: 'inline'
+      script: |
+        git config --global user.email "cibuild@steeltoe.com"
```
Comment (TimHess) [inline build/pr-code-cleanup.yml:78]:
```suggestion
        git config --global user.email "cibuild@steeltoe.io"
```
We don't own steeltoe.com

**Review by TimHess — APPROVED** [review]
LGTM - @ccheetham please take a look when you have time, we can enhance it post-merge if you catch anything

**Review by TimHess — DISMISSED** [review]

**Review by TimHess — APPROVED** [review]

**Pull request merged**